From ab972b02a2a0cc5d578c98faa3d0742e3bbb8770 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Konrad=20Zapa=C5=82owicz?= <konrad.zapalowicz@canonical.com>
Date: Tue, 4 Apr 2017 16:22:33 +0200
Subject: [PATCH 2/5] Fix hciattach on RPi3

This patch fixes the hciattach on Raspberry Pi 3 by applying the
following changes:

* don't set UART speed before loading firmware (thanks to
  https://github.com/MilhouseVH)
* change FIRMWARE_DIR to /lib/formware

These changes originated from LP: #1674509
---
 tools/hciattach_bcm43xx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/hciattach_bcm43xx.c b/tools/hciattach_bcm43xx.c
index 5c9c50e3..c1a005d8 100644
--- a/tools/hciattach_bcm43xx.c
+++ b/tools/hciattach_bcm43xx.c
@@ -377,6 +377,10 @@ int bcm43xx_init(int fd, int def_speed, int speed, struct termios *ti,
 			perror("Can't set host baud rate");
 			return -1;
 		}
+		/* give a moment to the controller to acknowledge the above
+		 * commands. without waiting it did fail on reset sometimes
+		 */
+		sleep(1);
 
 		if (bcm43xx_reset(fd))
 			return -1;
-- 
2.25.1

